test/cbmc/proofs/Posix_OtaStopTimer/Posix_OtaStopTimer_harness.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file Posix_OtaStopTimer_harness.c
+ * @brief Implements the proof harness for Posix_OtaStopTimer function.
+ */
+/*  POSIX includes for OTA library. */
+#include "ota_os_posix.h"
+
+void Posix_OtaStopTimer_harness()
+{
+    OtaTimerId_t otaTimer;
+
+    /* The valid range of values for OtaTimerId_t enum are [0,2).*/
+    __CPROVER_assume( otaTimer >= 0 && otaTimer < 2 );
+
+    Posix_OtaStopTimer( otaTimer );
```

Comment:
Shouldn't we create and start timer before stopping it?

Reply:
We do not require the timer to be created and started before stopping the timer. The `NONDET_STATIC` flag in the Makefile will assign values to `pOtaTimers` variable non-deterministically. This variable is passed to `timer_settime` function which does not require timer to be created and started since we are assuming it to be memory-fail proof. Additionally I have added a stub for the `timer_settime` function.